api/core/rag/datasource/vdb/tidb_vector/tidb_vector.py

```diff
@@ -0,0 +1,214 @@
+        return Table(
+            self._collection_name,
+            self._orm_base.metadata,
+            Column('id', String(36), primary_key=True, nullable=False),
+            Column("vector", VectorType(dim), nullable=False, comment="" if self._distance_func is None else f"hnsw(distance={self._distance_func})"),
+            Column("text", TEXT, nullable=False),
+            Column("meta", JSON, nullable=False),
+            Column("create_time", DateTime, server_default=sqlalchemy.text("CURRENT_TIMESTAMP")),
+            Column("update_time", DateTime, server_default=sqlalchemy.text("CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP")),
+            extend_existing=True
+        )
+
+    def __init__(self, collection_name: str, config: TiDBVectorConfig, distance_func: str = 'cosine'):
+        super().__init__(collection_name)
+        self._client_config = config
+        self._url = (f"mysql+pymysql://{config.user}:{config.password}@{config.host}:{config.port}/{config.database}?"
+                     f"ssl_verify_cert=true&ssl_verify_identity=true")
+        self._distance_func = distance_func.lower()
+        self._engine = create_engine(self._url)
+        self._orm_base = declarative_base()
+        self._dimension = 1536
+
+    def create(self, texts: list[Document], embeddings: list[list[float]], **kwargs):
+        logger.info("create collection and add texts, collection_name: " + self._collection_name)
+        self._create_collection(len(embeddings[0]))
+        self.add_texts(texts, embeddings)
+        self._dimension = len(embeddings[0])
+        pass
+
+    def _create_collection(self, dimension: int):
+        logger.info("_create_collection, collection_name " + self._collection_name)
+        lock_name = 'vector_indexing_lock_{}'.format(self._collection_name)
+        with redis_client.lock(lock_name, timeout=20):
+            collection_exist_cache_key = 'vector_indexing_{}'.format(self._collection_name)
+            if redis_client.get(collection_exist_cache_key):
+                return
+            with Session(self._engine) as session:
+                session.begin()
+                drop_statement = sql_text(f"""DROP TABLE IF EXISTS {self._collection_name}; """)
+                session.execute(drop_statement)
+                create_statement = sql_text(f"""
+                    CREATE TABLE IF NOT EXISTS {self._collection_name} (
+                        id CHAR(36) PRIMARY KEY,
+                        text TEXT NOT NULL,
+                        meta JSON NOT NULL,
+                        vector VECTOR<FLOAT>({dimension}) NOT NULL COMMENT "hnsw(distance={self._distance_func})",
+                        create_time DATETIME DEFAULT CURRENT_TIMESTAMP,
+                        update_time DATETIME DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP
+                    );
+                """)
+                session.execute(create_statement)
+                # TODO: create tidb-vector index
```

Comment:
there is a TODO task : create tidb-vector index @Weaxs

Reply:
I ask TiDB Vector developer.

`CREATE TABLE` with `COMMENT "hnsw(distance={self._distance_func})` is a temporary solution for setting vecotor index and distance func.

And TiDB Vector is not support to set index and distance by `CREATE INDEX/ADD INDEX` now, it will be supportted in the near future.

So ` # TODO: create tidb-vector index` is unneccessary now, maybe delete it ?